lib/internal/streams/pipeline.js

```diff
@@ -102,25 +106,52 @@ async function* fromReadable(val) {
 }
 
 async function pump(iterable, writable, finish) {
-  if (!EE) {
-    EE = require('events');
-  }
   let error;
+  let callback = noop;
+  const resume = (err) => {
+    if (!error && err) {
+      error = err;
+    }
+    const _callback = callback;
+    callback = noop;
+    _callback();
+  };
+  const onClose = () => {
+    resume(new ERR_STREAM_PREMATURE_CLOSE());
+  };
+
+  const waitForDrain = () => new Promise((resolve) => {
+    assert(callback === noop);
```

Comment:
@VoltrexMaster This is a how code comment looks. Once it fixed (which it is now) we can simply "resolve" it and it is hidden. Making it easier for other reviewers.